src/framework/auth/services/token/token.ts

```diff
@@ -1,25 +1,33 @@
 import { urlBase64Decode } from '../../helpers';
 
-export interface NbAuthToken {
-  getValue(): string;
-  isValid(): boolean;
-  toString(): string;
+export abstract class NbAuthToken {
+  abstract getValue(): string;
+  abstract isValid(): boolean;
+  abstract toString(): string;
+
+  getName(): string {
+    return (this.constructor as NbAuthTokenClass).NAME;
+  }
 }
 
-export interface NbTokenClass {
-  new (raw: string): NbAuthToken
+export interface NbAuthTokenClass {
+  NAME: string;
```

Comment:
I don't think **NAME** is a good name for this para. Maybe it has to be a sort of id?

Reply:
it's fine I suppose, as token.getId() would be confusing, expecting identity id rather than id of token type.